tests/api/test_polls.py

```diff
@@ -0,0 +1,69 @@
+from datetime import timedelta
+
+import pytest
+from django.utils import timezone
+
+from activities.models import Post
+from core.ld import format_ld_date
+
+
+@pytest.mark.django_db
+def test_get_poll(api_token, client):
```

Comment:
You can refactor these to use the new `api_client` that is now in `main`. Means you don't need the `HTTP_` headers in every call.

Reply:
oh, nice

Reply:
Done, but just as a note to check later, I noticed that if I don't put the `content_type="application/json"` to the post requests, instead of just failing the execution segfaults, so this is something to take a look at to see if it also segfaults a running server.

Reply:
It segfaults?! In Python? I'd be very curious to see why - if you could maybe open a separate issue with the details/logs, I'd like to get that fixed in Hatchway.

Reply:
Yup, just removed the content-type from the post requests in `test_polls.py` and then:

```
➜ pytest tests/api/test_polls.py              
======================================================= test session starts ========================================================
platform linux -- Python 3.11.1, pytest-7.2.1, pluggy-1.0.0
django: settings: takahe.settings (from option)
rootdir: /home/humrochagf/Projects/takahe/takahe, configfile: setup.cfg
plugins: anyio-3.6.2, asyncio-0.20.3, dotenv-0.5.2, django-4.5.2, httpx-0.21.3
asyncio: mode=Mode.STRICT
collected 2 items                                                                                                                  

tests/api/test_polls.py Fatal Python error: Segmentation fault

Current thread 0x00007f9e6b625b80 (most recent call first):
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/views/debug.py", line 232 in get_traceback_frame_variables
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/views/debug.py", line 544 in get_exception_traceback_frames
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/views/debug.py", line 490 in get_traceback_frames
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/views/debug.py", line 320 in get_traceback_data
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/views/debug.py", line 403 in get_traceback_text
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/log.py", line 125 in emit
  File "/home/humrochagf/.pyenv/versions/3.11.1/lib/python3.11/logging/__init__.py", line 978 in handle
  File "/home/humrochagf/.pyenv/versions/3.11.1/lib/python3.11/logging/__init__.py", line 1706 in callHandlers
  File "/home/humrochagf/.pyenv/versions/3.11.1/lib/python3.11/logging/__init__.py", line 1644 in handle
  File "/home/humrochagf/.pyenv/versions/3.11.1/lib/python3.11/logging/__init__.py", line 1634 in _log
  File "/home/humrochagf/.pyenv/versions/3.11.1/lib/python3.11/logging/__init__.py", line 1518 in error
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/log.py", line 241 in log_response
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 144 in response_for_exception
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 58 in inner
  File "/home/humrochagf/Projects/takahe/takahe/users/middleware.py", line 32 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/api/middleware.py", line 26 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/core/middleware.py", line 51 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/core/middleware.py", line 25 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django_htmx/middleware.py", line 45 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/deprecation.py", line 136 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/deprecation.py", line 136 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/debug_toolbar/middleware.py", line 49 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/deprecation.py", line 136 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/deprecation.py", line 136 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/deprecation.py", line 136 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/deprecation.py", line 136 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/whitenoise/middleware.py", line 60 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/deprecation.py", line 136 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/utils/deprecation.py", line 136 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/exception.py", line 56 in inner
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/core/handlers/base.py", line 140 in get_response
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/test/client.py", line 153 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/test/client.py", line 805 in request
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/test/client.py", line 541 in generic
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/test/client.py", line 441 in post
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/django/test/client.py", line 852 in post
  File "/home/humrochagf/Projects/takahe/takahe/tests/api/test_polls.py", line 12 in test_get_poll
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/python.py", line 195 in pytest_pyfunc_call
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/python.py", line 1789 in runtest
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 167 in pytest_runtest_call
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 260 in <lambda>
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 339 in from_call
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 259 in call_runtest_hook
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 220 in call_and_report
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 131 in runtestprotocol
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/runner.py", line 112 in pytest_runtest_protocol
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/main.py", line 349 in pytest_runtestloop
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/main.py", line 324 in _main
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/main.py", line 270 in wrap_session
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/main.py", line 317 in pytest_cmdline_main
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 167 in main
  File "/home/humrochagf/Projects/takahe/takahe/.venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 190 in console_main
  File "/home/humrochagf/Projects/takahe/takahe/.venv/bin/pytest", line 8 in <module>

Extension modules: pydantic.typing, pydantic.errors, pydantic.version, pydantic.utils, pydantic.class_validators, pydantic.config, pydantic.color, pydantic.datetime_parse, pydantic.validators, pydantic.networks, pydantic.types, pydantic.json, pydantic.error_wrappers, pydantic.fields, pydantic.parse, pydantic.schema, pydantic.main, pydantic.dataclasses, pydantic.annotated_types, pydantic.decorator, pydantic.env_settings, pydantic.tools, pydantic, charset_normalizer.md, lxml._elementpath, lxml.etree, markupsafe._speedups, psycopg2._psycopg, google._upb._message, _cffi_backend, PIL._imaging, yaml._yaml (total: 32)
[1]    71640 segmentation fault (core dumped)  pytest tests/api/test_polls.py
```